--- libmfcr2-0.0.3/mfcr2.c	2007-12-02 10:03:11.000000000 +0000
+++ libmfcr2-patched/mfcr2.c	2007-12-02 10:02:55.000000000 +0000
@@ -3569,6 +3569,12 @@
     if (call->state != UC_STATE_OFFERED)
         return  UC_RET_BAD_STATE;
     /*endif*/
+
+    /* BLOCK BRAZILIAN COLLECT CALLS - By Robert Almeida */
+    if (mfcr2->call->callparms.calling_party_category==9) {
+        return  UC_RET_BAD_DEVICE;
+    }
+
     if (mfcr2->call->callparms.call_type == UC_CALL_TYPE_NO_CHARGE_CALL)
     {
         mfcr2->mfc_state = MFC_SENT_ACCEPTED_NO_CHARGE;
